doc/cpp_domdec.rst

```diff
@@ -0,0 +1,186 @@
+
+        * group the cells into *cell groups* of the same kind of cell;
+        * assign each cell group to either a CPU core or GPU on a specific MPI rank.
+
+    The number of cells in each cell group depends on different factors,
+    including the type of the cell, and whether the cell group will run on a CPU
+    core or the GPU.
+
+    See :cpp:class:`arb::domain_decimposition`.
+
+Load balancer
+    A distributed algorithm that determines the domain decomposition using the
+    model recipe and and description of the available computational resources as
+    inputs.
+
+    See :cpp:func:`arb::partition_load_balance`.
+
+Hardware
+--------
+
+.. cpp:namespace:: arb::hw
+
+.. cpp:class:: node_info
+
+    Information about the computational resources available to a simulation, typically a
+    subset of the resources available on a physical hardware node.
+    When used for distributed simulations, where a model will be distributed over more than
+    one node, a :cpp:class:`hw::node_info` represents the resources available to the local
+    MPI rank.
+
+    .. container:: example-code
+
+        .. code-block:: cpp
+
+            // Make node that uses one thread for each available hardware thread,
+            // and one GPU if any GPUs are available.
+            hw::node_info node;
+            node.num_cpu_cores = threading::num_threads();
+            node.num_gpus = hw::num_gpus()>0? 1: 0;
+
+    .. cpp:function:: node_info() = default
+
+        Default constructor (sets 1 CPU core and 0 GPUs).
+
+    .. cpp:function:: node_info(unsigned cores, unsigned gpus)
+
+        Constructor that sets the number of :cpp:var:`cores` and :cpp:var:`gpus` available.
+
+    .. cpp:member:: unsigned num_cpu_cores = 1
+
+        The number of CPU cores available.
+
+        By default it is assumed that there is one core available.
+
+    .. cpp:member:: unsigned num_gpus = 0
+
+        The number of GPUs available.
+
+        By default it is assumed that there are no GPUs.
+
+Load Balancers
+--------------
+
+Load balancing generates a :cpp:class:`domain_decomposition` given a :cpp:class:`recipe`
+and a description of the hardware on which the model will run. Currently Arbor provides
+one load balancer, :cpp:func:`partition_load_balance`, and more will be added over time.
+
+If the model is distributed with MPI, the partitioning algorithm for cells is
+distributed with MPI communication. The returned :cpp:class:`domain_decomposition`
+describes the cell groups on the local MPI rank.
+
+.. Note::
+    Power users know their model and the hardware resources that
+    they are running on very well, and know best how to partition their model
+    workload to get optimal performance.
+    The :cpp:class:`domain_decomposition` type is kept very simple, and is
+    independent of any load balancing algorithm, so power users can supply their
+    own domain decomposition without using one of the built-in load balancers.
+
```

Comment:
I wouldn't use the term 'power user' — rather, I think it would be better to state that in cases where the provided load balancers are inadequate, one can provide a `domain_decomposition` object constructed by hand to better suit the model.